src/PrestaShopBundle/Entity/Repository/RequestSqlRepository.php

```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * Get count of all request sql's
+     *
+     * @return int  Number of request sql rows
+     */
+    public function getCount()
+    {
+        $statement = $this->connection->query("SELECT COUNT(rs.id_request_sql) AS c FROM $this->requestSqlTable rs");
+        $row = $statement->fetch();
+
+        return (int) $row['c'];
+    }
+
+    /**
+     * Find all filtered request sql's
+     *
+     * @param array $filters
+     *
+     * @return array
+     */
+    public function findByFilters(array $filters)
+    {
+        $qb = $this->connection->createQueryBuilder();
+
+        $conditionValues = array_filter($filters['filters'], function ($value) {
+            return !empty($value);
+        });
+
+        $qb->select('rs.*')
+            ->from($this->requestSqlTable, 'rs')
+            ->orderBy('`'.$filters['orderBy'].'`', $filters['sortOrder'])
+            ->setFirstResult($filters['offset'])
+            ->setMaxResults($filters['limit']);
+
+        $scalarFilters = array_filter($conditionValues, function ($key) {
+            return in_array($key, ['name', 'sql']);
+        }, ARRAY_FILTER_USE_KEY);
+
+        foreach ($scalarFilters as $column => $value) {
+            $qb->andWhere("`$column` LIKE :$column");
+            $qb->setParameter($column, '%'.$value.'%');
+        }
+
+        if (!empty($conditionValues['id_request_sql'])) {
+            $qb->andWhere('rs.id_request_sql = :id_request_sql');
+            $qb->setParameter('id_request_sql', $conditionValues['id_request_sql']);
+        }
+
+        $statement = $qb->execute();
+
+        return $statement->fetchAll(PDO::FETCH_ASSOC);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria = null)
+    {
+        $searchQueryBuilder = $this->buildQueryBySearchCriteria($searchCriteria);
+
+        $searchQueryBuilder
+            ->select('rs.*')
+            ->orderBy($searchCriteria->getOrderBy(), $searchCriteria->getOrderWay())
+            ->setFirstResult($searchCriteria->getOffset())
+            ->setMaxResults($searchCriteria->getLimit());
+
+        return $searchQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria = null)
+    {
+        $countQueryBuilder = $this->buildQueryBySearchCriteria($searchCriteria);
+        $countQueryBuilder->select('COUNT(rs.id_request_sql)');
+
+        return $countQueryBuilder;
+    }
+
+    /**
+     * Build partial query by search criteria
+     *
+     * @param SearchCriteriaInterface $criteria
+     *
+     * @return QueryBuilder
+     */
+    private function buildQueryBySearchCriteria(SearchCriteriaInterface $criteria)
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb->from($this->requestSqlTable, 'rs');
+
+        foreach ($criteria->getFilters() as $filterName => $value) {
+            if (empty($value)) {
```

Comment:
Why continue if filter is 0, or ''?

Reply:
if you submit empty filters and add empty string to sql condition, you will get no results.
Regarding '0' value, im not sure, maybe i can use `'' !== trim($value)`?